### DISCUSSION TITLE
[ПЛИ] Тип Bundle для кодирования протокола лаб исследования

### DISCUSSION THREAD

**Discussion by ai-fhir** [fhir-ru/core#53] [Лабораторная РГ]
# Вариант 1 (type = document) (Предложение Medlinx)

Протокол лабораторного исследования кодируется как Bundle с типом `document`

https://www.hl7.org/fhir/documents.html 

FHIR resources can be used to build documents that represent a composition: a coherent set of information that is a statement of healthcare information, including clinical observations and services. A document is an immutable set of resources with a fixed presentation that is authored and/or attested by humans, organizations and devices.

Использование этого формата накладывает дополнительные требования:

All documents have the same structure: a Bundle of resources of type "document" that has a Composition resource as the first resource in the bundle, followed by a series of other resources, referenced from the Composition resource, that provide supporting evidence for the document. The bundle gathers all the content of the document into a single XML or JSON document which may be signed and managed as required. The resources include both human readable and computer processable portions. In addition, the bundle may include CSS stylesheets , Provenance statements and a signature.

В данном подходе тип документа будет кодироваться в `Composition.type`

# Вариант 2 (type = collection) (Предложение ЦНИИОИЗ)

Протокол лабораторного исследования кодируется как Bundle с типом `collection`

Тип документа кодируется в `identifier`:

```
"resourceType": "Bundle",
    ...
    "identifier": [
        {
            "system": "urn:oid:1.2.643.5.1.13.13.12.2.77.8312.100.1.1.51",
            "value": "7854321",
            "use": "official",
            "type": {
                "coding": [
                    {
                        "system": "urn:CodeSystem:ru-identifier-type",
                        "code": "DOC_ID",
                        "display": "Уникальный идентификатор документа в МИС"
                    }
                ]
            }
        },
        {
            "system": "urn:oid:1.2.643.5.1.13.13.99.2.267",
            "value": "1.2.643.5.1.13.2.7.5.1.7.3",
            "type": {
                "coding": [
                    {
                        "system": "urn:CodeSystem:ru-identifier-type",
                        "code": "TEMPLATE_ID",
                        "display": "Уникальный идентификатор Шаблона документа"
                    }
                ]
            }
        },
        {
            "system": "urn:oid:1.2.643.5.1.13.13.11.1522",
            "value": "Протокол лабораторного исследования",
            "type": {
                "coding": [
                    {
                        "system": "urn:oid:1.2.643.5.1.13.13.11.1522",
                        "code": "7",
                        "display": "Протокол лабораторного исследования"
                    }
                ]
            }
        }
    ],
...
```

**Comment by Vrodvi** [comment]
> Тип документа кодируется в identifier:

Тип документа не может кодироваться в identifier, т.к. identifier - это уникальный идентификатор одного конкретного bundle ("Persistent identifier for the bundle").
Тип документа может кодироваться в элементах Composition.type и Composition.category

**Comment by Vrodvi** [comment]
В телеграм-чате "Лаб результаты в FHIR в РЭМД" совместно с ЦНИИОИЗ уже обсуждалась разработка протокола лабораторного исследования на FHIR.

Даже были подготовлены заготовки руководства по реализации:
![image](https://user-images.githubusercontent.com/92928137/144260310-0a7d78ca-50ce-4fd6-93a4-7dadeafc5d45.png)

К сожалению, указанные ссылки сейчас не работают.

Мне кажется будет правильным вынести основные решения и наработки из предыдущего обсуждения этой темы в данную карточку.

**Reply by ai-fhir** [reply]
Да, я принимал участие в разработке этого документа, но мы начинаем новый IG потому что:

1. Старый документ ставил своей целью повторить структуру заголовка CDA документа, а сейчас мы хотим полноценно закодировать всю структуру документа
1. На новый документ есть реальный запрос у ЦНИИОИЗ и есть шанс, что новые профили будут реально использоваться
1. За два года у нас обновился состав пользователей FHIR (кажется, что появились новые пользователи FHIR) и многие не принимали участие в прошлом обусждении.

Ссылки не работают, но вся работа по старым документам (профили, протоколы встреч, текст IG) доступна в версионном контроле.

Я конечно поддерживаю желание переиспользовать или опираться, где это возможно, на старые наработки.
